app/components/inputs/InputComponent.js

```diff
@@ -0,0 +1,194 @@
+    const { inputUnitDiv } = this.state;
+    var updatedInputUnitDiv = inputUnitDiv;
+    this.setState({ inputUnitDiv: updatedInputUnitDiv.classList.remove("active") });
+    onBlur && onBlur(e);
+  };
+  onKeyDown = (e) => {
+    const { onKeyDownSubmit, onKeyDown } = this.props;
+    (e.keyCode === 13) && onKeyDownSubmit && onKeyDownSubmit(e);
+    onKeyDown && !e.defaultPrevented && onKeyDown(e);
+  }
+  onKeyDown2 = (e) => {
+    if (e.keyCode === 13) {
+      e.preventDefault();
+    }
+    const { onKeyDownSubmit, onKeyDown } = this.props;
+    (e.keyCode === 13) && onKeyDownSubmit && onKeyDownSubmit(e);
+    onKeyDown && !e.defaultPrevented && onKeyDown(e);
+  }
+  onChange = () => {
+    var number =
+      (document.getElementById("input-main").innerHTML)
+      + (document.getElementById("input-sub").innerHTML);
+    document.getElementById("parsedInputDisplay").innerHTML = number;
+    document.getElementById("parsedInputDisplay").style.display = "inherit";
+    document.getElementById("viewInputDisplay").style.display = "none";
+  };
+  onChange2 = () => {
+    var number = document.getElementById("parsedInputDisplay").innerHTML;
+    var number_f = parseFloat(number);
+    var left_part = number_f.toFixed(2);
+    var right_part_raw = number_f - left_part;
+    var right_part_rounded = parseFloat(String(right_part_raw.toFixed(10)).slice(0, -1));
+
+    var right_part = right_part_rounded * 100000000;
+
+    // Used to compare value later
+    var prepad = right_part_rounded * 100000000;
+
+    right_part = right_part.toFixed(0);
+
+    // Deduce rounding errors
+    if (Math.floor(prepad) < right_part && Math.floor(prepad) == (parseInt(right_part)) - 1 ) {
+      right_part = (parseInt(right_part)) - 1;
+    }
+
+    // Deduce rounding errors (upto 10 decimal places)
+    else if (Math.floor(right_part_raw * 100000000) < right_part && Math.floor(right_part_raw * 100000000) == (parseInt(right_part)) - 1) {
+      right_part = (parseInt(right_part)) - 1;
+    }
+
+    // Pad for numbers, so 0032 decimal will not turn to 32
+    right_part = pad(String(right_part), 6);
+
+    if (right_part <= 0) {
+      right_part = "000000";
+    }
+
+    // Reset back to numbers
+    if (isNaN(left_part) == true || isNaN(right_part) == true) {
+      left_part = "0.00";
+      right_part = "000000";
+    }
+
+    document.getElementById("input-main").innerHTML = left_part;
+    document.getElementById("input-sub").innerHTML = right_part;
+
+    // Send to other older previous unstyled component and trigger event
+    document.getElementById("parsedInput").value = number_f;
+    var event = new Event("input", { bubbles: true });
+    var x = document.getElementById("parsedInput");
+    var tracker = x._valueTracker;
+    tracker.setValue("'" + x.value + "'");
+    x.dispatchEvent(event);
+
+    document.getElementById("parsedInputDisplay").style.display = "none";
+    document.getElementById("viewInputDisplay").style.display = "inherit";
```

Comment:
NAK on this. As a general rule in react projects, if you're writing code that uses `getElementById`, you're doing it wrong. 

You need to pass a semantic state or prop value to the component, then on the render() method you set the style (or how we usually do it, a className) to the components being rendered.

Reply:
Not picturing it if its not too much work can you give an example by doing it, would help out understanding a lot by exactly knowing how to do it in future.

Reply:
Here's one of the simplest examples I could find in our codebase: 

https://github.com/decred/decrediton/blob/master/app/components/shared/CopyToClipboard.js

It uses both props and state to change the css classes applied to html elements.

Reply:
Ah I see thank you, updated it.
Note I had to leave two instances to mirror the changes for validation and submit event.